### ISSUE TITLE
🐛[BUG]: PME reciprocal-space forces are wrong with geometry-dependent charges (CUDA)

### ISSUE THREAD

**Issue by sklenard** [NVIDIA/nvalchemi-toolkit-ops#115]
### Version

0.4.0-rc

### Describe the issue

`pme_reciprocal_space` gives the wrong gradient with respect to `positions` when charges are geometry-dependent (e.g. from an MLIP) on GPU with nvalchemiops 0.4.0-rc. Note that the gradient is correct on CPU.

The chain-rule term $(\partial E/\partial q)(\partial q/\partial R)$ gets counted twice on the position gradient when $q = q(R)$. This happens in `_PMEReciprocalCachedFirstGrad.backward()`'s CUDA fallback path, which recomputes gradients using the original (not detached) `charges` tensor:

$$
F_i^{\text{buggy}} = -\left[ \frac{\partial E}{\partial R_i}\bigg|_q + 2\sum_j \frac{\partial E}{\partial q_j}\frac{\partial q_j}{\partial R_i} \right].
$$

The example below compares the direct autograd forces with the manual calculation of the different contributions.

### Minimum reproducible example

```python
import torch

from nvalchemiops.torch.interactions.electrostatics import (
    estimate_ewald_parameters,
    estimate_pme_parameters,
    ewald_reciprocal_space,
    generate_k_vectors_ewald_summation,
    generate_k_vectors_pme,
    pme_reciprocal_space,
)

KE_EVA = 14.399645351950548  # Coulomb's constant, eV * Angstrom

DEVICE = "cuda"
DTYPE = torch.float64
ACCURACY = 1e-8
N_ATOMS = 256
BOX_LENGTH = 15.0

torch.manual_seed(0)


def build_system():
    cell = (torch.eye(3, dtype=DTYPE, device=DEVICE) * BOX_LENGTH).unsqueeze(0)
    positions = torch.rand(N_ATOMS, 3, dtype=DTYPE, device=DEVICE) * BOX_LENGTH
    return positions, cell


def charges_fn(positions):
    """Charges as a smooth function of positions, so dq/dpos != 0."""
    x, y, z = positions[:, 0], positions[:, 1], positions[:, 2]
    k = 2.0 * torch.pi / BOX_LENGTH
    charges = 0.5 * (torch.sin(k * x) + torch.cos(k * y) + torch.sin(k * z))
    return charges - charges.mean()  # neutralize


def check_force(label, recip_fn, recip_kwargs, positions):
    # (1) autograd, with charges = charges_fn(positions) -- the normal way
    pos = positions.detach().clone().requires_grad_(True)
    q = charges_fn(pos)
    E = recip_fn(positions=pos, charges=q, **recip_kwargs)
    (grad,) = torch.autograd.grad((E * KE_EVA).sum(), pos)
    F_autograd = -grad

    # (2) manual chain rule -- ground truth, doesn't rely on the op's
    # autograd composition at all, only on its own dE/dR and dE/dq.
    pos_leaf = positions.detach().clone().requires_grad_(True)
    q_leaf = charges_fn(pos_leaf.detach()).detach().clone().requires_grad_(True)
    E_leaf = recip_fn(positions=pos_leaf, charges=q_leaf, **recip_kwargs)
    dEdR, dEdq = torch.autograd.grad((E_leaf * KE_EVA).sum(), [pos_leaf, q_leaf])

    pos_q = positions.detach().clone().requires_grad_(True)
    q_again = charges_fn(pos_q)
    (chain_term,) = torch.autograd.grad(q_again, pos_q, grad_outputs=dEdq)
    F_manual = -(dEdR + chain_term)

    rel = ((F_manual - F_autograd).norm() / F_manual.norm()).item()
    print(f"--- {label} ---")
    print(f"  F_autograd = {F_autograd.norm().item():.6e}   F_manual = {F_manual.norm().item():.6e}")
    print(f"  rel diff   = {rel:.3e}")


positions, cell = build_system()

pme_params = estimate_pme_parameters(positions, cell, accuracy=ACCURACY)
ewald_params = estimate_ewald_parameters(positions, cell, accuracy=ACCURACY)

k_vectors_pme, k_squared_pme = generate_k_vectors_pme(cell, mesh_dimensions=pme_params.mesh_dimensions)
k_vectors_ewald = generate_k_vectors_ewald_summation(cell, k_cutoff=ewald_params.reciprocal_space_cutoff)

check_force(
    "PME", pme_reciprocal_space,
    dict(cell=cell, mesh_dimensions=pme_params.mesh_dimensions,
         k_vectors=k_vectors_pme, k_squared=k_squared_pme, alpha=pme_params.alpha),
    positions,
)
check_force(
    "Ewald", ewald_reciprocal_space,
    dict(cell=cell, k_vectors=k_vectors_ewald, alpha=ewald_params.alpha),
    positions,
)
```

### Relevant log output

```shell
--- PME ---
  F_autograd = 3.222603e+02   F_manual = 2.155225e+02
  rel diff   = 4.991e-01
--- Ewald ---
  F_autograd = 2.155225e+02   F_manual = 2.155225e+02
  rel diff   = 1.467e-17
```

### Environment details

```shell
+ GPU: H100
+ CUDA version: 12.8.0
+ Python version: 3.12.13
+ PyTorch version: 2.11.0
+ NVIDIA Warp version: 1.13.0
```
